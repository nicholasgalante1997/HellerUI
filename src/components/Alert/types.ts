import { WithCustomStyles } from '../../@types';

export interface NativeAlertProps {
  className?: string;
  id?: string;
  variant?: 'info' | 'warning' | 'error' | 'success'; //string union type
  title: string;
  isVisible: boolean;
  onClose: () => void;
}

export type AlertProps = WithCustomStyles<NativeAlertProps>;
