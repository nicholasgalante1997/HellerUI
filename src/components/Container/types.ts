import React from 'react';
import { SafeReactFC, WithCustomStyles } from '../../@types';

export type NativeContainerProps = {
    radius?: 'rounded' | 'normal' | 'none';
    background?: string;
    gradient?: {
        to: string;
        from: string;
        flow?: string;
    };
    width?: string | number;
    height?: string | number;
    opacity?: number;
    padding?: string;
    margin?: string;
    image?: string;
    asGridParent?: boolean;
    asGridChild?: boolean;
    colSpan?: number;
    onClick?: React.MouseEventHandler<HTMLDivElement>;
    onMouseEnter?: React.MouseEventHandler<HTMLDivElement>;
    onMouseLeave?: React.MouseEventHandler<HTMLDivElement>;
} & Partial<Pick<HTMLDivElement,
    'id' |
    'className' |
    'ariaLabel'
>>

export type ContainerProps =
    SafeReactFC<WithCustomStyles<NativeContainerProps>, HTMLDivElement>;
