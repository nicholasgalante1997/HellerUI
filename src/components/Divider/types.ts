export type HellerDividerProps = {
    fadeColor: string;
    focusColor: string;
};