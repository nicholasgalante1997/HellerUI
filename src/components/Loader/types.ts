export type LoaderProps = {
  fillColor?: string;
  mainColor?: string;
  spinRate?: number;
  height?: string;
  width?: string;
};
