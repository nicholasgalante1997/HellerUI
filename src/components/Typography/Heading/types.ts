import { Properties } from 'csstype';
import { SafeReactFC } from '../../../utils';

export interface StyledHeaderProps {
    color?: string;
    scale?: 100 | 200 | 300 | 400 | 500 | 600 | 700 | 800;
}

export interface NativeHeadingProps {
    as?: 'h1' | 'h2' | 'h3' | 'h4' | 'h5' | 'h6';
    customStyles?: Properties;
}

export type HeadingProps = SafeReactFC<NativeHeadingProps & StyledHeaderProps>;
