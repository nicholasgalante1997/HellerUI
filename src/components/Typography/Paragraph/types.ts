import { SafeReactFC, WithCustomStyles } from '../../../@types';

export type NativeParagraphProps = {
  color?: string;
  fontSize?: number;
  bold?: boolean;
  italic?: boolean;
  thin?: boolean;
};

export type ParagraphProps = WithCustomStyles<
  SafeReactFC<NativeParagraphProps>
> & { dataTestId?: string };
