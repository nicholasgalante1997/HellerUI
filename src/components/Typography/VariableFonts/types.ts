import { SafeReactFC, WithCustomStyles } from 'src/@types';
import { fontMap } from 'src/globals/fonts';

export type TextElementKeyType =
  | 'h1'
  | 'h2'
  | 'h3'
  | 'h4'
  | 'h5'
  | 'h6'
  | 'p'
  | 'span'
  | 'i'
  | 'b'
  | 'code';

export type VarFontProps = {
  fontKey: keyof typeof fontMap;
  implementation: TextElementKeyType;
  weight?:
    | '100'
    | '200'
    | '300'
    | '400'
    | '500'
    | '600'
    | '700'
    | '800'
    | '900';
} & SafeReactFC<WithCustomStyles<{}>>;
