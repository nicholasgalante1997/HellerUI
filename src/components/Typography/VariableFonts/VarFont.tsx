import React from 'react';

import fontBlob from '../../../fontBlob.json';

import { GoogleFontUrlBuilder } from '../../../@types';
import { useExtraneousFont } from '../../../utils/font/loader';
import { Loader } from '../../Loader';
import { getTextElement } from './VarFont.styles';
import { VarFontProps } from './types';

const VarFont = (props: VarFontProps) => {
  const {
    fontKey,
    implementation,
    children,
    customStyles,
    accessiblility,
    ...rest
  } = props;

  let mutableAccessibleChildren: JSX.Element[] | undefined = undefined;

  const fontData = React.useMemo(() => fontBlob[fontKey], [fontKey]);
  const fontInstance = new GoogleFontUrlBuilder(
    fontData.family,
    fontData.availableFontStyleVariants,
    fontData.meta
  );
  const { failed, loading } = useExtraneousFont(fontInstance.build());

  if (loading) {
    <Loader />;
  }

  if (failed) {
    console.error('[FATAL] link creation failed. falling back to local font.');
    return <div id="null-link-exception" data-dev="check-console" />;
  }

  if (accessiblility?.bionic && typeof children === 'string') {
    mutableAccessibleChildren = children.split(' ').map((word) => {
      const wordLength = word.length;
      const boldCharCount = Math.ceil(wordLength / 2);
      const boldSubString = word.substring(0, boldCharCount);
      const trailingSubString = word.substring(boldCharCount);
      return (
        <>
          <span style={{ fontWeight: 'bold' }}>{boldSubString}</span>
          {trailingSubString}&nbsp;
        </>
      );
    });
  }

  const Component = React.useMemo(
    () => getTextElement(implementation),
    [implementation]
  );
  return (
    <Component
      {...rest}
      style={{ ...customStyles, fontFamily: fontInstance.family }}
    >
      {mutableAccessibleChildren ?? children}
    </Component>
  );
};

export default VarFont;
