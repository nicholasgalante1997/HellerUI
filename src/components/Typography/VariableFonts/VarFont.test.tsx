import React from 'react';
import { render } from '@testing-library/react';
import VarFont from './VarFont';
import { TextElementKeyType } from './types';

describe('VariableFont', () => {
  test('default', () => {
    const { container } = render(
      <VarFont fontKey="Sarala - v10" implementation="h2">
        chief and dumpling
      </VarFont>
    );
    expect(container).toMatchSnapshot();
  });
  test('forwards styles', () => {
    const style = { fontStyle: 'italic' };
    const { container, queryByTestId } = render(
      <VarFont
        data-testid="varfont"
        customStyles={style}
        fontKey="Sarala - v10"
        implementation="h2"
      >
        chief and dumpling
      </VarFont>
    );
    expect(queryByTestId('varfont')).toHaveStyle(style);
    expect(container).toMatchSnapshot();
  });
  test('implementations', () => {
    const impls: readonly TextElementKeyType[] = [
      'b',
      'code',
      'h1',
      'h2',
      'h3',
      'h4',
      'h5',
      'h6',
      'p',
      'i',
      'span'
    ] as const;
    for (const i of impls) {
      const { container, unmount } = render(
        <VarFont
          data-testid="varfont"
          fontKey="Sarala - v10"
          implementation={i}
        >
          chief and dumpling
        </VarFont>
      );
      expect(container).toMatchSnapshot();
      unmount();
    }
  });
});
