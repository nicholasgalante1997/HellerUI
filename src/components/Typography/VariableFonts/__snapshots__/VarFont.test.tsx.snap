// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VariableFont default 1`] = `
<div>
  <h2
    class="sc-dkzDqf"
    style="font-family: Sarala;"
  >
    chief and dumpling
  </h2>
</div>
`;

exports[`VariableFont forwards styles 1`] = `
<div>
  <h2
    class="sc-dkzDqf"
    data-testid="varfont"
    style="font-style: italic; font-family: Sarala;"
  >
    chief and dumpling
  </h2>
</div>
`;

exports[`VariableFont implementations 1`] = `
<div>
  <b
    class="sc-papXJ"
    data-testid="varfont"
    style="font-family: Sarala;"
  >
    chief and dumpling
  </b>
</div>
`;

exports[`VariableFont implementations 2`] = `
<div>
  <p
    class="sc-iBkjds"
    data-testid="varfont"
    style="font-family: Sarala;"
  >
    chief and dumpling
  </p>
</div>
`;

exports[`VariableFont implementations 3`] = `
<div>
  <h1
    class="sc-gsnTZi"
    data-testid="varfont"
    style="font-family: Sarala;"
  >
    chief and dumpling
  </h1>
</div>
`;

exports[`VariableFont implementations 4`] = `
<div>
  <h2
    class="sc-dkzDqf"
    data-testid="varfont"
    style="font-family: Sarala;"
  >
    chief and dumpling
  </h2>
</div>
`;

exports[`VariableFont implementations 5`] = `
<div>
  <h3
    class="sc-hKMtZM"
    data-testid="varfont"
    style="font-family: Sarala;"
  >
    chief and dumpling
  </h3>
</div>
`;

exports[`VariableFont implementations 6`] = `
<div>
  <h4
    class="sc-eCYdqJ"
    data-testid="varfont"
    style="font-family: Sarala;"
  >
    chief and dumpling
  </h4>
</div>
`;

exports[`VariableFont implementations 7`] = `
<div>
  <h5
    class="sc-jSMfEi"
    data-testid="varfont"
    style="font-family: Sarala;"
  >
    chief and dumpling
  </h5>
</div>
`;

exports[`VariableFont implementations 8`] = `
<div>
  <h6
    class="sc-gKXOVf"
    data-testid="varfont"
    style="font-family: Sarala;"
  >
    chief and dumpling
  </h6>
</div>
`;

exports[`VariableFont implementations 9`] = `
<div>
  <p
    class="sc-iBkjds"
    data-testid="varfont"
    style="font-family: Sarala;"
  >
    chief and dumpling
  </p>
</div>
`;

exports[`VariableFont implementations 10`] = `
<div>
  <i
    class="sc-ftvSup"
    data-testid="varfont"
    style="font-family: Sarala;"
  >
    chief and dumpling
  </i>
</div>
`;

exports[`VariableFont implementations 11`] = `
<div>
  <span
    class="sc-kDDrLX"
    data-testid="varfont"
    style="font-family: Sarala;"
  >
    chief and dumpling
  </span>
</div>
`;
